tests/src/kakarot/instructions/test_system_operations.py

```diff
@@ -0,0 +1,229 @@
+        def test__exec_auth__should_fail_invalid_invoker(
+            self, cairo_run, private_key, invoker_address, message
+        ):
+            invoker_address += 1
+            serialized_msg, msg_hash, stack, memory = prepare_stack_and_memory(
+                invoker_address, message, private_key
+            )
+
+            msg_hash_int = int.from_bytes(msg_hash, "big")
+
+            with SyscallHandler.patch(
+                "Kakarot_evm_to_starknet_address", invoker_address, 0x1234
+            ), SyscallHandler.patch(
+                "ICairo1Helpers.library_call_keccak",
+                lambda class_hash, data: list(int_to_uint256(msg_hash_int)),
+            ):
+                stack, evm = cairo_run(
+                    "test__auth",
+                    stack=stack,
+                    memory=memory,
+                    invoker_address=invoker_address,
+                )
+
+            assert stack == ["0x0"]
+            assert evm["message"]["authorized"] == {"is_some": 0, "value": 0}
+
+    class TestAuthCall:
+        @SyscallHandler.patch("IERC20.balanceOf", lambda addr, data: [0, 1])
+        @SyscallHandler.patch("IAccount.get_nonce", lambda addr, data: [NONCE])
+        @SyscallHandler.patch("IAccount.bytecode", lambda addr, data: [0])
+        @SyscallHandler.patch(
+            "Kakarot_cairo1_helpers_class_hash",
+            CAIRO1_HELPERS_CLASS_HASH,
+        )
+        def test__auth_then_authcall_should_return_correct_evm_frame(
+            self, cairo_run, private_key, invoker_address, message
+        ):
+            serialized_msg, msg_hash, auth_stack, auth_memory = (
+                prepare_stack_and_memory(invoker_address, message, private_key)
+            )
+
+            gas = [100000, 0]
+            called_address = 0xC411
+            value = [100, 0]
+            args_offset = [0, 0]
+            args_len = [0, 0]
+            ret_offset = [0, 0]
+            ret_len = [0, 0]
+            authcall_stack = [
+                gas,
+                list(int_to_uint256(called_address)),
+                value,
+                args_offset,
+                args_len,
+                ret_offset,
+                ret_len,
+            ]
+
+            msg_hash_int = int.from_bytes(msg_hash, "big")
+            with SyscallHandler.patch(
+                "Kakarot_evm_to_starknet_address", invoker_address, 0x1234
+            ), SyscallHandler.patch(
+                "ICairo1Helpers.library_call_keccak",
+                lambda class_hash, data: list(int_to_uint256(msg_hash_int)),
+            ):
+                evm = cairo_run(
+                    "test__auth_authcall",
+                    auth_stack=auth_stack,
+                    auth_memory=auth_memory,
+                    invoker_address=invoker_address,
+                    authcall_stack=authcall_stack,
+                )
+
+            assert evm["message"]["authorized"] == {
+                "is_some": 0,
+                "value": 0,
```

Comment:
doesn't it fit in one line without dangling `,`?

Reply:
no